WordPress/gutenberg
# [RNMobile] Extract caption component

## Description
This extracts out a component for captions to be used by image and video blocks. Its use in image blocks should cause no behavior change. Its use in video blocks is a change from the current plain text caption. As a result:

1. Video blocks now have rich-text captions ([gutenberg-mobile issue](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1244)); and
2. The ["Virtual keyboard dismisses when focus moves from image/video caption to paragraph" issue](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1221) is resolved for video captions (image captions were addressed in a [previous PR](https://github.com/WordPress/gutenberg/pull/16759)).

This also adds the ability to select the video block when the video block's caption is selected without playing the video. This is important so that you can bring back the video block's edit buttons in the toolbar after selecting the caption, and I confirmed with @iamthomasbishop that this was desired behavior.

![Screen Shot 2019-07-30 at 3 32 39 PM](https://user-images.githubusercontent.com/4656348/62159275-4bc28080-b2df-11e9-87ae-627ba12e1f92.png)

## How has this been tested?
The behavior for the image block captions should be unchanged, and video blocks should now have rich text captions.

I tested this by following the testing scenarios from [the original PR adding rich text captions to the image block](https://github.com/WordPress/gutenberg/pull/14883) on both image and video captions.

The only issue with testing this on video blocks is that because currently [mobile video captions do not currently work on the web](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1237) and [web video captions do not work on mobile](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1152), I was not able to really test the video captions interop with web. The html for the captions looks good though. Interestingly, although the video block from mobile doesn't display the video itself on web, I was able to see the captions for the (undisplayed) videos on web. 🤷‍♂️

One additional scenario I tested is the selection of the video block from the caption:
1. Select the caption for a video block
2. Observe rich text editing buttons in the toolbar _without_ the pencil editing icon for the video block
3. Tap on the video above the caption
4. Observe that:
  1. The rich text editing buttons are removed from the toolbar;
  2. The pencil icon for editing the video block is added to the toolbar;
  3. The video does **not** start playing; and
  4. The keyboard is dismissed

## Checklist:
